common/scripts/copyright-linter.js

```diff
@@ -26,6 +29,8 @@ const oldCopyrightBanner = RegExp(
 if (filePaths) {
   filePaths.forEach((filePath) => {
     let fileContent = fs.readFileSync(filePath, { encoding: "utf8" });
+    const copyrightBanner = getCopyrightBanner(/\r\n/.test(fileContent))
```

Comment:
perhaps instead of searching for `/\r\n/`, find the first `\n` and test if behind it is a `\r`?
In the case where the file is unix style (LF, no CR/`\r`), `\r\n` will search the entire text, while searching for `\n` will terminate early in both unix style and windows style endings, without searching the whole text.

Reply:
The lookahead regex would slower than a regular one but would definitely be faster than parsing the entire file.

Reply:
Good suggestion!  Trying that now and it clearly helps, but I don't think it explains enough of why the hook is still so slow, even for small files...  Gonna spend a couple more minutes optimizing it further...

Reply:
I didn't even think of using lookahead/lookbehind pattern, more like:
```js
const firstLineEnd = text.search(/\n/);
const crlf = text[firstLineEnd-1] == '\r';
```

Reply:
Finally getting back to this - just pushed that change